tests/integration/conftest.py

```diff
@@ -50,3 +53,44 @@ async def app(ops_test: OpsTest, app_name: str):
     await ops_test.model.wait_for_idle(status=ActiveStatus.name, idle_period=120)
 
     yield application
+
+
+@pytest_asyncio.fixture(scope="module")
+async def app_ip(app: Application):
+    """Get unit IP address and add to /etc/hosts."""
+    # Get the IP address which is in the status message
+    ip_regex = r"[0-9]+(?:\.[0-9]+){3}"
+    time.sleep(100)  # wait for ingress
```

Comment:
I wonder if `ops_test.model.wait_for_idle()` works here? It doesn't always, e.g. in the case of Jenkins as you might have seen. Otherwise:
1. Since this is an async function, would `await asyncio.sleep()` work it? It would release the event loop to potentially do other things
2. Do you think it is worth having some sort of check for an early exit? E.g., maybe a function that looks for whether the IP address has appeared in the status message and then stops the waiting? It would potentially reduce the time the tests take to execute

Reply:
Working on "Task was destroyed but it is pending!" even after successful integration tests.

Reply:
Possible fix here: https://github.com/canonical/charmed-magma-orchestrator/pull/44

Reply:
Didn't work. At least, it seems that doesn't have an impact on tests.